version/version.go

```diff
@@ -1,5 +1,29 @@
 package version
 
+import (
+	"fmt"
+	"runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	sdkVersion "github.com/operator-framework/operator-sdk/version"
+)
+
+var log = logf.Log.WithName("version")
+
+//var needs to be used instead of const for ldflags
 var (
-	Version = "0.0.1"
+	Version            = "0.0.1+git"
```

Comment:
Why the `+git`?

Reply:
Added it so that it helps us identify that we have built the operator directly from git and not using any release. We will add Version and GitCommit while releasing through ldflags.

Reply:
we can do 
```
GitCommit=`git rev-parse --short HEAD`
Version=1.0.0+$GitCommit

go build -ldflags "-X version.Version=$Version"
```
to inject git commit hash during build time

Reply:
Although, imo I won't go with adding `+githash` while building the operator everytime, as doing that will stop us from identifying how the binary was built (was it built directly from branch? what if someone has made changes and not committed before building binary). I would reserve `+githash` for releases. 

or 

We can go with `version+githash` for building and reserve `GitCommit` to identify the release.

Reply:
>what if someone has made changes and not committed before building binary). I would reserve +githash for releases.

You can detect that there are uncommitted changes and add a `dirty`or some such suffix. Not sure there is any reason to reserve the git hash for releases.

Reply:
> Added it so that it helps us identify that we have built the operator directly from git and not using any release.

Even for a release we will be building from some git repo.

> We will add Version and GitCommit while releasing through ldflags.

Where is that being done?